pkg/migrate/cstor/pool.go

```diff
@@ -0,0 +1,468 @@
+		LabelSelector: string(apis.StoragePoolClaimCPK) + "=" + c.SPCObj.Name,
+	})
+	if err != nil {
+		return err
+	}
+	if c.SPCObj.Spec.BlockDevices.BlockDeviceList == nil {
+		if c.SPCObj.Spec.MaxPools == nil {
+			return errors.Errorf("invalid spc %s neither has bdc list nor maxpools", c.SPCObj.Name)
+		}
+		if *c.SPCObj.Spec.MaxPools != len(cspList.Items) {
+			return errors.Errorf("maxpool count does not match csp count expected: %d got: %d",
+				*c.SPCObj.Spec.MaxPools, len(cspList.Items))
+		}
+		return nil
+	}
+	bdMap := map[string]int{}
+	for _, bdName := range c.SPCObj.Spec.BlockDevices.BlockDeviceList {
+		bdMap[bdName]++
+	}
+	for _, cspObj := range cspList.Items {
+		for _, rg := range cspObj.Spec.Group {
+			for _, bdObj := range rg.Item {
+				bdMap[bdObj.Name]++
+			}
+		}
+	}
+	for bdName, count := range bdMap {
+		// if bd is configured properly it should occur exactly twice
+		// one in spc spec and one in csp spec
+		if count != 2 {
+			return errors.Errorf("bd %s is not configured properly", bdName)
+		}
+	}
+	return nil
+}
+
+// getSPCWithMigrationStatus gets the spc by name and verifies if the spc is already
+// migrated or not. The spc will not be present in the cluster as the last step
+// of migration deletes the spc.
+func (c *CSPCMigrator) getSPCWithMigrationStatus(spcName string) (*apis.StoragePoolClaim, bool, error) {
+	spcObj, err := spc.NewKubeClient().
+		Get(spcName, metav1.GetOptions{})
+	// verify if the spc is already migrated. If an equivalent cspc exists then
+	// spc is already migrated as spc is only deleted as last step.
+	if k8serrors.IsNotFound(err) {
+		klog.Infof("spc %s not found.", spcName)
+		_, err = c.OpenebsClientset.CstorV1().
+			CStorPoolClusters(c.OpenebsNamespace).Get(spcName, metav1.GetOptions{})
+		if err != nil {
+			return nil, false, errors.Wrapf(err, "failed to get equivalent cspc for spc %s", spcName)
+		}
+		return nil, true, nil
+	}
+	if err != nil {
+		return nil, false, err
+	}
+	return spcObj, false, nil
+}
+
+// csptocspi migrates a CSP to CSPI based on hostname
+func (c *CSPCMigrator) cspTocspi(cspiObj *cstor.CStorPoolInstance) error {
+	var err1 error
+	hostnameLabel := types.HostNameLabelKey + "=" + cspiObj.Labels[types.HostNameLabelKey]
+	spcLabel := string(apis.StoragePoolClaimCPK) + "=" + c.CSPCObj.Name
+	cspLabel := hostnameLabel + "," + spcLabel
+	cspObj, err := getCSP(cspLabel)
+	if err != nil {
+		return err
+	}
+	if cspiObj.Annotations[types.OpenEBSDisableReconcileLabelKey] != "" {
+		klog.Infof("Migrating csp %s to cspi %s", cspObj.Name, cspiObj.Name)
+		err = c.scaleDownDeployment(cspObj, c.OpenebsNamespace)
+		if err != nil {
+			return err
+		}
+		// once the old pool pod is scaled down and bdcs are patched
+		// bring up the cspi pod so that the old pool can be renamed and imported.
+		cspiObj.Annotations[types.OpenEBSCStorExistingPoolName] = "cstor-" + string(cspObj.UID)
+		delete(cspiObj.Annotations, types.OpenEBSDisableReconcileLabelKey)
```

Comment:
Can update the Status.Phase to `Offline`? Because having `empty` (or) `Pending` status will try to create the pool. So if we update the Status.Phase to `Offline` will avoid getting into the creation path of the pool.